src/features/student/declaredPrograms/components/overlays/AddDeclaredProgramDrawer/AddDeclaredProgramDrawer.test.ts

```diff
@@ -0,0 +1,191 @@
+  }
+
+  const getCancelConfirmButtons = () => wrapper.findComponent(AvCancelConfirmButtonsStub)
+
+  beforeEach(async () => {
+    vi.clearAllMocks()
+    setActivePinia(createPinia())
+
+    const store = useDeclaredProgramsStore()
+    store.displayAddDeclaredProgramDrawer()
+
+    wrapper = mountComponent<typeof AddDeclaredProgramDrawer>(
+      AddDeclaredProgramDrawer,
+      {
+        global: {
+          stubs
+        }
+      },
+      {
+        usePinia: false
+      }
+    )
+
+    await wrapper.vm.$nextTick()
+  })
+
+  BddTest().when('the component is mounted', () => {
+    BddTest().then('it should render the drawer with correct props', () => {
+      const drawer = wrapper.findComponent({ name: 'AvDrawer' })
+
+      expect(drawer.exists()).toBe(true)
+      expect(drawer.props('position')).toBe('right')
+      expect(drawer.props('width')).toBe('40rem')
+    })
+
+    BddTest().then('it should render the title', () => {
+      const title = wrapper.find('h2')
+
+      expect(title.exists()).toBe(true)
+      expect(title.text()).toContain('Ajouter une formation déclarée')
+    })
+
+    BddTest().then('it should render accordion group with three accordions', () => {
+      const accordionsGroup = wrapper.findComponent({ name: 'AvAccordionsGroup' })
+      const accordions = wrapper.findAllComponents({ name: 'AvAccordion' })
+
+      expect(accordionsGroup.exists()).toBe(true)
+      expect(accordions).toHaveLength(3)
+    })
+
+    BddTest().then('it should render all form field components in first accordion', () => {
+      expect(wrapper.find('[data-testid="title-form-field"]').exists()).toBe(true)
+      expect(wrapper.find('[data-testid="description-form-field"]').exists()).toBe(true)
+      expect(wrapper.find('[data-testid="institution-form-field"]').exists()).toBe(true)
+      expect(wrapper.find('[data-testid="period-form-field"]').exists()).toBe(true)
+      expect(wrapper.find('[data-testid="result-form-field"]').exists()).toBe(true)
+      expect(wrapper.find('[data-testid="source-form-field"]').exists()).toBe(true)
+      expect(wrapper.find('[data-testid="link-form-field"]').exists()).toBe(true)
+    })
+
+    BddTest().then('it should render footer buttons', () => {
+      const cancelConfirmButtons = getCancelConfirmButtons()
+
+      expect(cancelConfirmButtons.exists()).toBe(true)
+      expect(cancelConfirmButtons.props('cancelLabel')).toBe('Annuler')
+      expect(cancelConfirmButtons.props('confirmLabel')).toBe('Enregistrer')
+    })
+
+    BddTest().then('it should render form element', () => {
+      const form = wrapper.find('form')
+      expect(form.exists()).toBe(true)
+    })
+  })
+
+  BddTest().when('store showAddDeclaredProgramDrawer is false', () => {
+    BddTest().then('it should pass false to drawer show prop', async () => {
+      const store = useDeclaredProgramsStore()
+      store.hideAddDeclaredProgramDrawer()
+      await wrapper.vm.$nextTick()
```

Comment:
```suggestion
  BddTest().when('store showAddDeclaredProgramDrawer is false', () => {
    beforeEach(() => {
      const store = useDeclaredProgramsStore()
      store.hideAddDeclaredProgramDrawer()
      await wrapper.vm.$nextTick()    
    })
    
    BddTest().then('it should pass false to drawer show prop', async () => {
```

Par souci de cohérence, ce when devrait être un and dans le when component is mounted et il devrait y avoir un beforeEach qui exécute le scénario du and.

Valable pour toute la suite du fichier de test.